src/main/java/org/springframework/data/elasticsearch/core/index/MappingParameters.java

```diff
@@ -323,6 +331,10 @@ public void writeTypeAndParametersTo(XContentBuilder builder) throws IOException
 			builder.field(FIELD_PARAM_POSITIVE_SCORE_IMPACT, positiveScoreImpact);
 		}
 
+		if (dims >= 1) {
+			builder.field(FIELD_PARAM_DIMS, dims);
+		}
+
```

Comment:
According to https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html `dims` is a required parameter. So it should be checked that if `type == FieldType.DenseVector` then the `dims` value must be greater or equal 1.

Reply:
That makes sense! I combined the two validations for `dims` [here](https://github.com/spring-projects/spring-data-elasticsearch/pull/1708) and [here](https://github.com/spring-projects/spring-data-elasticsearch/pull/1708/files#diff-5d797228d21b667cac484c820a8b903ebc13b603af75cc41cf4018ebab5d081aR204) and added a new [test](https://github.com/spring-projects/spring-data-elasticsearch/pull/1708/files#diff-8244d8ceae8257120f809f9edeecfdb56f37dc9a573c315b7afd5068d7019ccfR84). I thought it would be best to group it with the other validations, so that malformed data does not make it past the creation of `MappingParameters`.